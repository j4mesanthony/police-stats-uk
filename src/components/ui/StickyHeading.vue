<script setup lang="ts">
import { computed } from 'vue';
import { useWindowSize } from '@/global/composables/useWindowSize';
import { useScrollPosition } from '@/global/composables/useScrollPosition'; 

const { isMobile } = useWindowSize();
const { yPos } = useScrollPosition();
const scrollThreshold = computed(() => isMobile.value ? 60 : 25);
</script>

<template>
    <h2 class="sticky top-0 z-50 mb-6 text-2xl font-bold text-slate-50" :class="{ 'top-[67px]': isMobile, 'backdrop-blur-md transition-all border-b-[1px] pt-5 pb-5 border-slate-600': yPos > scrollThreshold }">
        <slot>Heading</slot>
    </h2>
</template>