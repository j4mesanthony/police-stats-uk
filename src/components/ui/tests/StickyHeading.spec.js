import { describe, it, expect } from 'vitest';
import StickyHeading from '../StickyHeading.vue';
import { createApp } from 'vue';
import { renderToString } from '@vue/server-renderer';

describe('StickyHeading', () => {
    it('should not regress (snapshot test)', async () => {
        const app = createApp(StickyHeading);
        const string = await renderToString(app);
        expect(string).toMatchSnapshot();
    })
})