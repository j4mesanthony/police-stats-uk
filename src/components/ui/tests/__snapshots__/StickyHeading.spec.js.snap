// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StickyHeading > should not regress (snapshot test) 1`] = `"<h2 class=\\"sticky top-0 z-50 mb-6 text-2xl font-bold text-slate-50\\"><!--[-->Heading<!--]--></h2>"`;
